smooth/components/component_fuel_cell_chp.py

```diff
@@ -156,3 +156,19 @@ def update_flows(self, results, sim_params):
         # Check if the component has an attribute 'flows', if not, create it as an empty dict.
         Component.update_flows(self, results, sim_params, self.name + '_electric')
         Component.update_flows(self, results, sim_params, self.name + '_thermal')
+
+    def update_costs(self, results, sim_params):
+        # Get the name of the flow of this component.
+        flow_name = list(self.flows)
+        # Get the amount of electric energy supplied by the chp this interval time step [Wh].
+        this_energy_supplied_electric = self.flows[flow_name[1]][sim_params.i_interval]
+        # Call the function of the mother component to save costs and art. costs for this run.
+        Component.update_costs(self, results, sim_params, this_energy_supplied_electric)
+
+    def update_var_emissions(self, results, sim_params):
+        # Get the name of the flow of this component.
+        flow_name = list(self.flows)
+        # Get the amount of electric energy supplied by the chp this interval time step [Wh].
+        this_energy_supplied_electric = self.flows[flow_name[1]][sim_params.i_interval]
+        # Call the function of the mother component to save emissions for this run.
+        Component.update_var_emissions(self, results, sim_params, this_energy_supplied_electric)

```

Comment:
I added it, because the update functions need to know the dependant value in order to use it, otherwise the dependant value will be 0 all the time.

Still, I thought in case it is a methan fuel cell it could have emissions ... this will only happen if model settings provides a var_emissions value for this component, otherwise they stay 0

For the cost, it is the same concerning the dependant value, but I think I have to delete it, because the costs are taken into account at other places: e. g. H2 is without cost when we generate it ourselves, and in case we had connection to a methan/H2 grid, the cost should be modeled in that component.

Reply:
https://github.com/rl-institut/smooth/pull/15/commits/df7abcf4ba958837fc976f07a666d13ad7bd5cbb

Reply:
deleted update_cost() https://github.com/rl-institut/smooth/pull/15/commits/5e3f211514d7d60db83d73f3683ae0ae9dcbb65a

Reply:
update variable cost and emission now only in component.py 250967a61c4ccb6e20a062b3cc584a6ab7841d58